dist/DateRangePicker.js

```diff
@@ -130,11 +130,28 @@ var DateRangePicker = _react2['default'].createClass({
   componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
     var nextDateStates = this.getDateStates(nextProps);
     var nextEnabledRange = this.getEnabledRange(nextProps);
+    var yearMonth = this.getYearMonth(nextProps);
 
-    this.setState({
+    var updatedState = {
+      selectedStartDate: null,
+      hideSelection: false,
       dateStates: this.state.dateStates && _immutable2['default'].is(this.state.dateStates, nextDateStates) ? this.state.dateStates : nextDateStates,
       enabledRange: this.state.enabledRange && this.state.enabledRange.isSame(nextEnabledRange) ? this.state.enabledRange : nextEnabledRange
-    });
+    };
+
+    // if numberOfCalendars are not 2 update as normal
+    if (yearMonth && nextProps.numberOfCalendars !== 2) {
+      updatedState.year = yearMonth.year;
+      updatedState.month = yearMonth.month;
+
+      // prevent updating month & year if there are 2 calendars and the date falls in the 2nd month
+      // this will prevent the calendar moving if within the 2 months already visiable
+    } else if (this.isYearMonthNotCurrentlyDisplayed(yearMonth, nextProps.numberOfCalendars)) {
```

Comment:
To remove the duplicate `updatedState` assignment, and the `numberOfCalendars` count assertion

Maybe we could write something like -

```js
if (yearMonth && !this.isYearMonthVisible(yearMonth, nextProps.numberOfCalendars)) {
  updatedState.year = yearMonth.year;
  updatedState.month = yearMonth.month;
} 

this.setState(updatedState);
```

And then maybe update `isYearMonthNotCurrentlyDisplayed` to be re-implemented in a truthy way, something like -

```js
isYearMonthVisible: function isYearMonthVisible(yearMonth, numberOfCalendars) {
  const isSameYear = (yearMonth.year === this.state.year);
  const isMonthVisible =  (yearMonth.month === this.state.month) || (numberOfCalendars === 2 && (yearMonth.month - 1 === this.state.month));
 
  return  isSameYear && isMonthVisible;
}
```

Note: Code hasn't been tested.